pkg/api/api.go

```diff
@@ -65,8 +66,8 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 
 // RegisterFlagsWithPrefix adds the flags required to config this to the given FlagSet with the set prefix.
 func (cfg *Config) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet) {
-	f.StringVar(&cfg.AlertmanagerHTTPPrefix, prefix+"http.alertmanager-http-prefix", "/alertmanager", "HTTP URL path under which the Alertmanager ui and api will be served.")
-	f.StringVar(&cfg.PrometheusHTTPPrefix, prefix+"http.prometheus-http-prefix", "/prometheus", "HTTP URL path under which the Prometheus api will be served.")
+	f.StringVar(&cfg.AlertmanagerHTTPPrefix, path.Join(prefix, "http.alertmanager-http-prefix"), "/alertmanager", "HTTP URL path under which the Alertmanager ui and api will be served.")
+	f.StringVar(&cfg.PrometheusHTTPPrefix, path.Join(prefix, "http.prometheus-http-prefix"), "/prometheus", "HTTP URL path under which the Prometheus api will be served.")
```

Comment:
**nit:** These are flags, no URL routes. The prefix is meant to be concatenated as it currently is.